### PR TITLE
add message type byte prefix to messages

### PR DESCRIPTION
**Pull request by noot**
Thank you for contributing to Mina! Please see `CONTRIBUTING.md` if you haven't
yet. In that doc, there are more details around how to start our CI.

based off https://github.com/MinaProtocol/mina/pull/8721

Explain your changes here.
- add `MessageType` byte to `sendStreamMsgMsg` and `publishMsg`
- add prefix before sending msg, remove prefix when reading from stream

Explain how you tested your changes here.
`go test ./...`

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

Closes #0000
Closes #0000

### CODE REVIEW

**Comment by nholland94** [comment]
@noot I notice that there are a few commits missing from https://github.com/MinaProtocol/mina/pull/7724 (`606a369` onward). Do we want to pull those in?

**Comment by nholland94** [comment]
!ci-build-me

**Comment by nholland94** [comment]
!ci-build-me

File: src/app/libp2p_helper/src/libp2p_helper/main.go [hunk 1]
```diff
@@ -55,6 +55,7 @@ type streamState int
 const (
 	STREAM_DATA_UNEXPECTED streamState = iota
 	STREAM_DATA_EXPECTED
+	STREAM_QUERY_HANDLED
```
Comment (georgeee) [inline src/app/libp2p_helper/src/libp2p_helper/main.go:58]: @nholland94 any idea why was this change introduced?

**Comment by georgeee** [comment]
Opened other PR #9083 for these changes

File: src/app/libp2p_helper/src/libp2p_helper/main.go (hunk 1 above)
Comment (nholland94) [reply src/app/libp2p_helper/src/libp2p_helper/main.go:58]: Looks like a bad merge. I renamed these enums in a previous PR, and this is one of the old names. The only 2 enums we need here are the `STREAM_DATA_*` ones.

**Pull request closed without merging**